base/strings/basic.jl

```diff
@@ -62,6 +62,7 @@ convert(::Type{Array{UInt8}}, s::AbstractString) = String(s).data
 convert(::Type{String}, s::AbstractString) = String(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
 convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
+convert(::Type{String}, s::Symbol) = unsafe_wrap(String, Cstring(s))
```

Comment:
There's probably not enough of a performance difference to matter.  I'll go with the "safer" unsafe version.

```jl
julia> symbols = [gensym() for i=1:10^6];
julia> f1(symbols) = for s in symbols unsafe_wrap(String, Cstring(s)) end
julia> f2(symbols) = for s in symbols unsafe_string(Cstring(s)) end

julia> @time f1(symbols)
  0.040287 seconds (2.00 M allocations: 91.553 MB, 14.31% gc time)

julia> @time f2(symbols)
  0.052429 seconds (2.00 M allocations: 106.812 MB, 15.83% gc time)
```